hash.c

```diff
@@ -1731,11 +1731,13 @@ rb_hash_replace(VALUE hash, VALUE hash2)
  *     hsh.size      ->  integer
  *
  *  Returns the number of key-value pairs in the hash.
+ *  <code>Hash#length</code> and <code>Hash#size</code> are both equivalent to
+ *  each other.
  *
  *     h = { "d" => 100, "a" => 200, "v" => 300, "e" => 400 }
  *     h.length        #=> 4
  *     h.delete("a")   #=> 200
- *     h.length        #=> 3
+ *     h.size          #=> 3
```

Comment:
I don't really have strong opinions on this, but I could imagine someone else would attempt to replace `#size` with `#length `. I would say leave the `h.length` and just add `h.size`.

Reply:
Thanks @yuki24 for your feedback. I like the way you propose. I have updated the commit with your suggestion and rebased it against trunk.